index.js

```diff
@@ -89,6 +95,83 @@ async function create ({
   }
 }
 
+/**
+ * peersFullAck
+ *
+ * @description check if there is at least one peer fully synchronized (up to date) with our drive
+ * @async
+ * @param {Object} contentFeed hypercore content feed
+ * @param {Bitfield} localBitfield a copy of the local contents bitfield
+ * @param {Map} visitedStats every time a new peer-ack arrives, if the local bitfield hast the value, we mark the visitedStats removing the entry from the map.
+ */
+async function peersFullAck (contentFeed, localBitfield, visitedStats) {
+  if (!contentFeed) {
+    throw new Error('content feed is required')
+  }
+
+  // note: consider add a timeout to cancel the wait
+  const result = {}
+  result.promise = new Promise((resolve, reject) => {
+    result.resolve = () => {
+      contentFeed.off('peer-ack', checkSync)
+      resolve()
+    }
+
+    result.reject = err => {
+      contentFeed.off('peer-ack', checkSync)
+      reject(err)
+    }
+  })
+
+  function checkSync (_, have) {
+    if (localBitfield.get(have.start)) {
+      // mark visited
+      visitedStats.delete(have.start)
+      if (visitedStats.size === 0) {
```

Comment:
Interesting approach! Could we maybe add a `console.log` saying the file got uploaded?